### PR TITLE
feat: add wsa to ppm report

### PR DESCRIPTION
**Pull request by GaryJBlake**
In order to have a good experience with our community, we recommend that you read the [contributing guidelines](https://github.com/vmware-samples/power-validated-solutions-for-cloud-foundation/blob/main/CONTRIBUTING.md) for making a pull request.

**Summary of Pull Request**

- Added Workspace ONE Access to Password Policy Manager Reports
- Added `Publish-WsaDirectoryPasswordPolicy` cmdlet to retrieve password policies for Workspace ONE Access Directory Users and output the details.
- Added `Publish-WsaLocalPasswordPolicy` cmdlet to retrieve password policies for Workspace ONE Access Local Users and output the details.

Signed-off-by: Gary Blake <gblake@vmware.com>

**Type of Pull Request**

- [ ] This is a bug fix.
- [x] This is an enhancement or feature.
- [ ] This is a code style / formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is something else.

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

- [x] Tests have been completed (for bug fixes / features).
- [ ] Documentation has been added / updated (for bug fixes / features).

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### CODE REVIEW

File: PowerValidatedSolutions.psm1 [hunk 1]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy PasswordComplexity -allDomains
+        This example will return password complexity policy for Workspace ONE Access Directory Users
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy AccountLockout -allDomains
+        This example will return account lockout policy for Workspace ONE Access Directory Users
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy PasswordExpiration -allDomains -drift -reportPath "F:\Reporting" -policyFile "passwordPolicyConfig.json"
+        This example will return password expiration policy for Workspace ONE Access Directory Users and compare the configuration against the passwordPolicyConfig.json
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20058]:
```suggestion
        This example will return the password expiration policy for Workspace ONE Access Local Directory Users and compare the configuration against the passwordPolicyConfig.json
```

File: PowerValidatedSolutions.psm1 [hunk 2]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy PasswordExpiration -allDomains -drift
+        This example will return password expiration policy for Workspace ONE Access Directory Users and compares the configuration against the product defaults
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20062]:
```suggestion
        This example will return the password expiration policy for Workspace ONE Access Local Directory Users and compares the configuration against the product defaults
```

File: PowerValidatedSolutions.psm1 [hunk 3]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+        .DESCRIPTION
+        The Publish-WsaDirectoryPasswordPolicy cmdlet retrieves the requested password policy for all ESXi hosts and converts
+        the output to HTML. The cmdlet connects to the SDDC Manager using the -server, -user, and -password values:
+        - Validates that network connectivity and authentication is possible to SDDC Manager
+        - Validates that network connectivity and authentication is possible to vCenter Server
+        - Retrieves the requested password policy for Workspace ONE Access Local Users and converts to HTML
+
+        .EXAMPLE
+        Publish-WsaLocalPasswordPolicy -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -policy PasswordExpiration -wsaFqdn sfo-wsa01.sfo.rainpole.io -wsaRootPass VMw@re1! -allDomains
+        This example will return password expiration policy for Workspace ONE Access Directory Users
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20119]:
```suggestion
        This example will return the password expiration policy for Workspace ONE Access Local Users
```

File: PowerValidatedSolutions.psm1 [hunk 4]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaLocalPasswordPolicy -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -policy PasswordComplexity -wsaFqdn sfo-wsa01.sfo.rainpole.io -wsaRootPass VMw@re1! -allDomains
+        This example will return password complexity policy for Workspace ONE Access Directory Users
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20123]:
```suggestion
        This example will return the password complexity policy for Workspace ONE Access Local Users
```

File: PowerValidatedSolutions.psm1 [hunk 5]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaLocalPasswordPolicy -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -policy AccountLockout -wsaFqdn sfo-wsa01.sfo.rainpole.io -wsaRootPass VMw@re1! -allDomains
+        This example will return account lockout policy for Workspace ONE Access Directory Users
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20127]:
```suggestion
        This example will return the account lockout policy for Workspace ONE Access Local Users
```

File: PowerValidatedSolutions.psm1 [hunk 6]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaLocalPasswordPolicy -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -policy PasswordExpiration -wsaFqdn sfo-wsa01.sfo.rainpole.io -wsaRootPass VMw@re1! -allDomains -drift -reportPath "F:\Reporting" -policyFile "passwordPolicyConfig.json"
+        This example will return password expiration policy for Workspace ONE Access Directory Users and compare the configuration against the passwordPolicyConfig.json
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20131]:
```suggestion
        This example will return the password expiration policy for Workspace ONE Access Local Users and compare the configuration against the passwordPolicyConfig.json
```

File: PowerValidatedSolutions.psm1 [hunk 7]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaLocalPasswordPolicy -server sfo-vcf01.sfo.rainpole.io -user admin@local -pass VMw@re1!VMw@re1! -policy PasswordExpiration -wsaFqdn sfo-wsa01.sfo.rainpole.io -wsaRootPass VMw@re1! -allDomains -drift
+        This example will return password expiration policy for Workspace ONE Access Directory Users and compares the configuration against the product defaults
```
Comment (tenthirtyam) [inline PowerValidatedSolutions.psm1:20135]:
```suggestion
        This example will return the password expiration policy for Workspace ONE Access Local Users and compares the configuration against the product defaults
```

**Review by tenthirtyam — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 8]
```diff
@@ -74,6 +74,8 @@
 - Added `Publish-NsxEdgeAccountLockout` cmdlet to retrieve account lockout policy for NSX Edge and output the details.
 - Added `Get-PasswordPolicyDefault` cmdlet to retrieve the default password policy settings either to an output or JSON file.
 - Added `Start-PasswordPolicyConfig` cmdlet to perform the configuration of password policy across a VMware Cloud Foundation instance.
+- Added `Publish-WsaDirectoryPasswordPolicy` cmdlet to retrieve password policies for Workspace ONE Access Directory Users and output the details.
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply CHANGELOG.md]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 9]
```diff
@@ -19456,24 +19456,43 @@ Function Request-WsaPasswordExpiration {
...
         Request-WsaPasswordExpiration -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1!
         This example retrieves the password expiration policy for Workspace ONE Access instance sfo-wsa01
+
+        .EXAMPLE
+        Request-WsaPasswordExpiration -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -drift -reportPath "F:\Reporting" -policyFile "passwordPolicyConfig.json"
+        This example retrieves the password expiration policy for Workspace ONE Access instance sfo-wsa01 and checks the configuration drift using the provided configuration JSON
+
+        .EXAMPLE
+        Request-WsaPasswordExpiration -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -drift
+        This example retrieves the password expiration policy for Workspace ONE Access instance sfo-wsa01 and compares the configuration against the product defaults
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:19466]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 10]
```diff
@@ -19499,27 +19518,48 @@ Function Request-WsaPasswordComplexity {
...
         Request-WsaPasswordComplexity -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1!
         This example retrieves the password complexity policy for Workspace ONE Access instance sfo-wsa01
+
+        .EXAMPLE
+        Request-WsaPasswordExpiration -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -drift -reportPath "F:\Reporting" -policyFile "passwordPolicyConfig.json"
+        This example retrieves the password complexity policy for Workspace ONE Access instance sfo-wsa01 and checks the configuration drift using the provided configuration JSON
+
+        .EXAMPLE
+        Request-WsaPasswordExpiration -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -drift
+        This example retrieves the password complexity policy for Workspace ONE Access instance sfo-wsa01 and compares the configuration against the product defaults
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:19528]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 3 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20119]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 4 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20123]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 5 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20127]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 6 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20131]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 7 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20135]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 11]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+        .DESCRIPTION
+        The Publish-WsaDirectoryPasswordPolicy cmdlet retrieves the requested password policy for Workspace ONE Access
+        and converts the output to HTML. The cmdlet connects to the SDDC Manager using the -server, -user, and
+        -password values:
+        - Validates that network connectivity and authentication is possible to Workspace ONE Access
+        - Retrieves the requested password policy for Workspace ONE Access and converts to HTML
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy PasswordExpiration -allDomains
+        This example will return password expiration policy for Workspace ONE Access Directory Users
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20046]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 12]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy PasswordComplexity -allDomains
+        This example will return password complexity policy for Workspace ONE Access Directory Users
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20050]: Fixed

File: PowerValidatedSolutions.psm1 (hunk 2 above)
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20062]: Fixed

File: PowerValidatedSolutions.psm1 [hunk 13]
```diff
@@ -19930,6 +20029,159 @@ Function Update-WsaLocalUserAccountLockout {
...
+
+        .EXAMPLE
+        Publish-WsaDirectoryPasswordPolicy -server sfo-wsa01.sfo.rainpole.io -user admin -pass VMw@re1! -policy AccountLockout -allDomains
+        This example will return account lockout policy for Workspace ONE Access Directory Users
```
[the remark this answers is not in the document]
Comment (GaryJBlake) [reply PowerValidatedSolutions.psm1:20054]: Fixed

**Pull request merged**